autorequests/autorequests.go

```diff
@@ -0,0 +1,57 @@
+package autorequests
+
+import (
+	"github.com/distributeddesigns/currency"
+)
+
+// AutoRequest :  A buy or sell request for a user
+type AutoRequest struct{ Amount, Trigger currency.Currency }
+
+// AutoRequestStore : Map stock -> user -> request
+type AutoRequestStore map[string](map[string]AutoRequest)
+
+// NewAutoRequestStore :
+func NewAutoRequestStore() *AutoRequestStore {
+	ars := make(AutoRequestStore)
+
+	return &ars
+}
+
+// AddBuyAmount :
+func (ars *AutoRequestStore) AddAutorequest(stock, userID string, amount currency.Currency) {
+	// Initialize the new user -> request map if don't find
+	// any entries for the stock in the store
+	if _, found := (*ars)[stock]; !found {
```

Comment:
merge this for now. Comment for posterity.

We should look into the right way to do dual maps. I've done it a different way in quoteserve.